WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,160 @@ Private.event_prototypes = {
           end
         ]=]
       end
-      return ret:format(trigger.currencyId or 1)
+
+      ret = ret .. [=[
+        -- Gather currency data for account characters
+        if currencyInfo and currencyInfo.isAccountTransferable then
+          if C_CurrencyInfo.IsAccountCharacterCurrencyDataReady() then
+            currencyInfo.accountQuantity = currencyInfo.quantity
+            currencyInfo.realAccountQuantity = currencyInfo.quantity
+            if currencyInfo.transferPercentage then
+                currencyInfo.realCharacterQuantity = math.floor(currencyInfo.quantity * (currencyInfo.transferPercentage/100))
+            end
+
+            accountCurrencyData = Private.ExecEnv.FetchCurrencyDataFromAccountCharacters(
+              currencyInfo.currencyID,
+              event == "CURRENCY_TRANSFER_LOG_UPDATE"
+            ) or {}
+
+            for _, currencyData in ipairs(accountCurrencyData) do
+              if currencyData.quantity then
+                local realQuantity = currencyData.quantity
+                if currencyInfo.transferPercentage and currencyInfo.transferPercentage > 0 and currencyInfo.transferPercentage < 100 then
+                  realQuantity = math.floor(realQuantity * (currencyInfo.transferPercentage/100))
+                end
+                currencyData.realCharacterQuantity = realQuantity
+                currencyInfo.realAccountQuantity = currencyInfo.realAccountQuantity + realQuantity
+                currencyInfo.accountQuantity = currencyInfo.accountQuantity + currencyData.quantity
+              end
+            end
+          else
+            C_CurrencyInfo.RequestCurrencyDataForAccountCharacters()
+          end
+        end
+
+        local active = true
+        local primaryCheckFailed = false
+
+        -- Check quantity-related values
+        for attribute, triggerValue in pairs(triggerQuantities) do
```

Comment:
This should be rewritten so that checking the triggerValue happens at function creation time, not every time the trigger runs.

So the looping should be outside of the string, and the string should only include a check if the triggerValue is not nil. This makes the resulting function much smaller in the usual case that there's no or one check.